push/service.go

```diff
@@ -2,20 +2,47 @@ package push
 
 import (
 	"encoding/json"
+	"fmt"
 
 	"github.com/RobotsAndPencils/buford/payload"
 	"github.com/RobotsAndPencils/buford/push"
 	"github.com/pkg/errors"
 	"golang.org/x/net/context"
+
+	"github.com/micromdm/micromdm/pubsub"
+	"github.com/micromdm/micromdm/queue"
 )
 
 type Push struct {
 	db      *DB
 	pushsvc *push.Service
 }
 
-func New(db *DB, push *push.Service) *Push {
-	return &Push{db, push}
+func New(db *DB, push *push.Service, sub pubsub.Subscriber) (*Push, error) {
+	pushSvc := Push{db, push}
+
+	commandQueuedEvents, err := sub.Subscribe("push-info", queue.CommandQueuedTopic)
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"subscribing push to %s topic", queue.CommandQueuedTopic)
+	}
+	go func() {
+		for {
+			select {
+			case event := <-commandQueuedEvents:
+				udid, _, err := queue.UnmarshalQueuedCommand(event.Message)
+				if err != nil {
+					fmt.Println(err)
+				}
+				_, err = pushSvc.Push(nil, udid)
```

Comment:
One thing I was thinking was an attribute or some signal on each queued command that tells whether to send a push or not. In this way its up to the command-sender to decide when pushes get sent out (including some way to trigger an arbitrary push say at the end of queuing a bunch of commands).